### PR TITLE
fix(mcp): sync User-Agent version with package.json and add CLI validation tests

### PR DESCRIPTION
**Pull request by aptsalt**
## Summary
- **Bug fix**: `userAgent.ts` hardcoded `VERSION = '0.3.1'` but `package.json` is at `0.3.3`, causing the User-Agent header to report an incorrect version to Stripe servers
- **Test coverage**: Added missing tests for `validateApiKey`, `validateStripeAccount`, and `--tools` deprecation handling

## Changes
- `tools/modelcontextprotocol/src/userAgent.ts`: Sync hardcoded version to `0.3.3` (with comment to keep in sync with `package.json`)
- `tools/modelcontextprotocol/src/test/index.test.ts`: Add 6 new test cases covering `validateApiKey` (4 cases), `validateStripeAccount` (2 cases), and `--tools` deprecation (1 case)

## Test plan
- [ ] Existing tests pass
- [ ] New validation tests cover edge cases
- [ ] User-Agent string now reports correct version

Related: Noticed during version bump in #330

### CODE REVIEW

File: tools/modelcontextprotocol/src/userAgent.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-const VERSION = '0.3.1';
+// Keep in sync with version field in package.json
+const VERSION = '0.3.3';
```
Comment (matv-stripe) [inline tools/modelcontextprotocol/src/userAgent.ts]: can we use the VERSION constant exported here https://github.com/stripe/ai/blob/main/tools/typescript/src/shared/constants.ts#L8 ?

**Review by matv-stripe — COMMENTED** [review]
thank you @aptsalt , left a small comment

File: tools/modelcontextprotocol/src/userAgent.ts (hunk 1 above)
Comment (aptsalt) [reply tools/modelcontextprotocol/src/userAgent.ts]: Good call! I looked into importing from `tools/typescript/src/shared/constants.ts` but since `@stripe/mcp` is a separate package (no dependency on the typescript toolkit), that import won't resolve.

Two options I see:

1. **Enable `resolveJsonModule` + import from own `package.json`** — but `rootDir: "./src"` in tsconfig blocks importing `../package.json`. Would need to adjust rootDir or add a build-time injection like tsup does for the shared package.

2. **Replicate the shared pattern** — use `process.env.PACKAGE_VERSION` and inject it at build time via the build script.

Which approach do you prefer? Happy to implement either. Option 2 seems most consistent with how the shared toolkit already handles it.

Comment (matv-stripe) [reply tools/modelcontextprotocol/src/userAgent.ts]: i like option 2! thank you!

**Comment by aptsalt** [comment]
Implemented Option 2! Here's what the new commit does:

**`src/userAgent.ts`** — Replaced hardcoded version with:
```typescript
const VERSION = process.env.PACKAGE_VERSION || '0.0.0-development';
```

**`scripts/inject-version.js`** — Post-`tsc` build step that reads `version` from `package.json` and replaces `process.env.PACKAGE_VERSION` in the compiled `dist/userAgent.js` with the actual version string. This mirrors how the shared toolkit handles it via tsup's `define` option.

**`package.json`** — Updated build script:
```
tsc && node scripts/inject-version.js && node -e "require(\"fs\").chmodSync(\"dist/index.js\", \"755\")"
```

**`jest.setup.ts`** + **`jest.config.ts`** — Sets `PACKAGE_VERSION` env var during tests so the version resolves correctly.

All 24 tests pass ✅, build injects version correctly (`"0.3.3"` baked into dist output).

File: tools/modelcontextprotocol/src/userAgent.ts (hunk 1 above)
Comment (aptsalt) [reply tools/modelcontextprotocol/src/userAgent.ts]: Done! Just pushed the implementation. userAgent.ts now reads from process.env.PACKAGE_VERSION which gets injected at build time by scripts/inject-version.js same pattern the shared toolkit uses via tsup's define. No more manual version syncing. All tests pass.

**Comment by aptsalt** [comment]
@matv-stripe I have updated the PR can you review it please

File: tools/modelcontextprotocol/scripts/inject-version.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const fs = require('fs');
+const path = require('path');
+
+const pkg = require('../package.json');
+const version = pkg.version;
+
+const userAgentPath = path.join(__dirname, '..', 'dist', 'userAgent.js');
+let content = fs.readFileSync(userAgentPath, 'utf8');
+content = content.replace(
+  'process.env.PACKAGE_VERSION',
```
Comment (matv-stripe) [inline tools/modelcontextprotocol/scripts/inject-version.js]: This feels brittle / hacky - can we use a build system?

**Review by matv-stripe — CHANGES_REQUESTED** [review]
sorry -just saw this!

File: tools/modelcontextprotocol/scripts/inject-version.js (hunk 2 above)
Comment (aptsalt) [reply tools/modelcontextprotocol/scripts/inject-version.js]: Done! Replaced `inject-version.js` with [tsup](https://tsup.egoist.dev/) — uses its `define` option to inline `process.env.PACKAGE_VERSION` at build time, same pattern as `tools/typescript/tsup.config.ts`. The post-build script and `scripts/` directory are removed. Build + all 24 tests pass.

**Review by matv-stripe — APPROVED** [review]
thank you!

**Comment by aptsalt** [comment]
@matv-stripe  All review feedback addressed, checks passing, CLA signed. Ready to merge when you get a chance! 🙏
